crates/sc-consensus-subspace/src/archiver.rs

```diff
@@ -243,6 +252,48 @@ where
     fn key(key_index: u16) -> Vec<u8> {
         (Self::KEY_PREFIX, key_index.to_le_bytes()).encode()
     }
+
+    /// Get segment headers that are expected to be included at specified block number.
+    pub fn segment_headers_for_block(&self, block_number: BlockNumber) -> Vec<SegmentHeader> {
+        let Some(last_segment_index) = self.max_segment_index() else {
+            return Vec::new(); // not initialized
+        };
+
+        // Special case for the initial segment (for genesis block).
+        if block_number == 1 {
+            return vec![self
+                .get_segment_header(SegmentIndex::ZERO)
+                .expect("The initial segment header must exist at this point.")];
```

Comment:
This is a bit better proof following above check for known last segment index:
```suggestion
                .expect("Segment headers are stored in monononically increasing order; qed")];
```

If there is a segment index present and we store monotnically increasing segment headers, then the first header exists, quod erat demonstrandum.